pkg/tfvars/internal/cache/cache.go

```diff
@@ -96,6 +96,21 @@ func cacheFile(reader io.Reader, filePath string, sha256Checksum string) (err er
 		}
 	}()
 
+	// Make sure that the temp file does not exist after completion of the function
+	defer func() {
+		_, err2 := os.Stat(tempPath)
+		if err2 == nil {
+			// If the file still exists, then something went wrong and we have to delete
+			// it manually.
+			err3 := os.Remove(tempPath)
+			if err3 != nil {
+				err = err3
```

Comment:
I don't like clobbering `err` unless it's `nil`.  I'm fine if you want to log an error you're about to discard though.

```go
if err2 := os.Remove(tempPath); err2 != nil && !os.IsNotExist(err2) {
  if err == nil {
    err = err2
  } else {
    logrus.Errorf("failed to clean up %s: %v", tempPath, err2)
  }
}
```
or some such.

Reply:
thanks! done as you said